### PR TITLE
feat: implement AuthContext with persistence and stay signed in

### PR DESCRIPTION
**Pull request by stephenhungg**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Create AuthContext.tsx in utils folder with AsyncStorage persistence
- Wrap app with AuthProvider in _layout.tsx
- Add login/signup integration with context
- Create dashboard page for authenticated users
- Implement stay signed in feature with checkbox
- Add smart routing based on authentication state
- Include loading states and error handling
- All code formatted with prettier

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

No

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

AuthContext.tsx

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

Idk

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

CURSOR!!!

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @rachaelch3n

### CODE REVIEW

**Comment by jeannineruiz** [comment]
WHY IS THIS PR SO HUGE

File: api/config.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+// API Configuration for Rails Backend
+const BASE_URL =
+  process.env.EXPO_PUBLIC_API_BASE_URL || 'http:/10.2.193.239:3000';
```
Comment (jeannineruiz) [inline api/config.ts]: is this supposed to be http://

**Comment by jeannineruiz** [comment]
Nice.

**Review by jeannineruiz — APPROVED** [review]

File: api/rails/client.ts
Comment (rachaelch3n) [inline api/rails/client.ts:1]: thanks for renaming

File: api/rails/queries/query.ts
Comment (rachaelch3n) [inline api/rails/queries/query.ts:1]: also thanks

File: src/app/auth/login/index.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  const [showPassword, setShowPassword] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [staySignedIn, setStaySignedIn] = useState(false);
+
+  const { login } = useAuth();
+
+  const handleLogin = async () => {
+    try {
+      setIsLoading(true);
+      console.log('Logging in with:', { email, password });
```
Comment (rachaelch3n) [inline src/app/auth/login/index.tsx]: get rid of this console log. dont want to leak sensitive info

File: api/config.ts (hunk 1 above)
Comment (rachaelch3n) [inline api/config.ts]: lets not hardcode our IP into the main branch. can you change the fallback to "http://localhost:3000" instead. your .env should hold your own IP address

File: src/app/_layout.tsx [hunk 3]
```diff
@@ -1,25 +1,16 @@
...
-      <Stack.Navigator screenOptions={{ headerShown: false }}>
-        <Stack.Screen name="Index" component={IndexScreen} />
-      </Stack.Navigator>
-    </SafeAreaProvider>
+    <AuthProvider>
+      <Stack>
+        <Stack.Screen name="index" options={{ headerShown: false }} />
+        <Stack.Screen name="auth/login" options={{ headerShown: false }} />
+        <Stack.Screen name="auth/signup" options={{ headerShown: false }} />
+        <Stack.Screen name="dashboard" options={{ headerShown: false }} />
```
Comment (rachaelch3n) [inline src/app/_layout.tsx]: can we get rid of the dashboard? it's a good way to check if signup is successful but we are going to change this later.

File: src/app/auth/login/index.tsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+      const loginData: DriverLoginData = {
+        email,
+        password,
+      };
+
+      await login(loginData, staySignedIn);
+      console.log('Login successful');
```
Comment (rachaelch3n) [inline src/app/auth/login/index.tsx]: get rid of this log

File: src/app/auth/login/index.tsx [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+      Alert.alert('Success', 'Welcome back!', [
+        { text: 'OK', onPress: () => router.push('/dashboard') },
+      ]);
+    } catch (error) {
+      console.error('Login error:', error);
```
Comment (rachaelch3n) [inline src/app/auth/login/index.tsx]: get rid of log

File: src/app/auth/signup/index.tsx [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+  const [password, setPassword] = useState('');
+  const [showPassword, setShowPassword] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const { signup } = useAuth();
+
+  const handleSignUp = async () => {
+    try {
+      setIsLoading(true);
+      console.log('Creating driver account...');
```
Comment (rachaelch3n) [inline src/app/auth/signup/index.tsx]: get rid of this

File: src/app/auth/signup/index.tsx [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+        email,
+        password,
+        password_confirmation: password,
+        first_name,
+        last_name,
+        phone: phone_number,
+      };
+
+      await signup(signupData);
+      console.log('Driver created successfully');
```
Comment (rachaelch3n) [inline src/app/auth/signup/index.tsx]: and this

File: src/app/auth/login/index.tsx (hunk 4 above)
Comment (rachaelch3n) [inline src/app/auth/login/index.tsx]: marking here too

File: src/app/auth/login/index.tsx (hunk 5 above)
Comment (rachaelch3n) [inline src/app/auth/login/index.tsx]: here

Comment (rachaelch3n) [inline src/app/auth/login/index.tsx]: console

File: src/app/auth/login/index.tsx (hunk 4 above)
Comment (rachaelch3n) [inline src/app/auth/login/index.tsx]: console

File: src/app/auth/login/index.tsx [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+
+      const loginData: DriverLoginData = {
+        email,
+        password,
+      };
+
+      await login(loginData, staySignedIn);
+      console.log('Login successful');
+      Alert.alert('Success', 'Welcome back!', [
+        { text: 'OK', onPress: () => router.push('/dashboard') },
```
Comment (rachaelch3n) [inline src/app/auth/login/index.tsx]: just alert, don't push to dashboard

File: src/app/auth/login/index.tsx [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+
+  return (
+    <SafeAreaView style={authStyles.container}>
+      <ScrollView contentContainerStyle={authStyles.content}>
+        <Text style={authStyles.logo}>Logo</Text>
+        <Text style={authStyles.title}>Welcome</Text>
+        <Text style={authStyles.subtitle}>Please input your details</Text>
+        <TextInput
+          style={authStyles.input}
+          placeholder="youremailaddress@address.com"
```
Comment (rachaelch3n) [inline src/app/auth/login/index.tsx]: this should be Email@gmail.com (following the designs)

File: src/utils/AuthContext.tsx [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+export function AuthProvider({ children }: { children: ReactNode }) {
+  const [driver, setDriver] = useState<DriverResponse | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const isAuthenticated = !!driver;
+
+  // Load stored driver data on app start
+  useEffect(() => {
+    const loadStoredDriver = async () => {
+      try {
+        console.log('Loading stored driver...');
```
Comment (rachaelch3n) [inline src/utils/AuthContext.tsx]: marking this to get rid of

File: src/utils/AuthContext.tsx [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+        const storedDriver = await AsyncStorage.getItem('driver');
+        console.log('Stored driver:', storedDriver);
```
Comment (rachaelch3n) [inline src/utils/AuthContext.tsx]: marking

File: src/utils/AuthContext.tsx [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+
+        if (storedDriver) {
+          const parsedDriver = JSON.parse(storedDriver);
+          console.log('Setting driver from storage:', parsedDriver);
```
Comment (rachaelch3n) [inline src/utils/AuthContext.tsx]: marking

File: src/utils/AuthContext.tsx [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+          setDriver(parsedDriver);
+        } else {
+          console.log('No stored driver found');
```
Comment (rachaelch3n) [inline src/utils/AuthContext.tsx]: marking

File: src/utils/AuthContext.tsx [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+        }
+      } catch (error) {
+        console.error('Failed to load stored driver:', error);
```
Comment (rachaelch3n) [inline src/utils/AuthContext.tsx]: marking

File: src/utils/AuthContext.tsx [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+      } finally {
+        setIsLoading(false);
+        console.log('Finished loading, isLoading set to false');
```
Comment (rachaelch3n) [inline src/utils/AuthContext.tsx]: marking

File: src/utils/AuthContext.tsx [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+
+    loadStoredDriver();
+  }, []);
+
+  const login = async (credentials: DriverLoginData, staySignedIn = false) => {
+    const driverData = await driverAPI.login(credentials);
+    setDriver(driverData);
+
+    if (staySignedIn) {
+      console.log('Saving driver to storage:', driverData);
```
Comment (rachaelch3n) [inline src/utils/AuthContext.tsx]: marking

File: src/utils/AuthContext.tsx [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+      await AsyncStorage.setItem('driver', JSON.stringify(driverData));
+    }
+  };
+
+  const signup = async (driverData: DriverSignupData, staySignedIn = false) => {
+    const newDriver = await driverAPI.signup(driverData);
+    setDriver(newDriver);
+
+    if (staySignedIn) {
+      console.log('Saving driver to storage:', newDriver);
```
Comment (rachaelch3n) [inline src/utils/AuthContext.tsx]: marking

File: src/utils/AuthContext.tsx [hunk 18]
```diff
@@ -0,0 +1,104 @@
...
+      await AsyncStorage.setItem('driver', JSON.stringify(newDriver));
+    }
+  };
+
+  const logout = async () => {
+    console.log('Logging out, clearing storage');
```
Comment (rachaelch3n) [inline src/utils/AuthContext.tsx:77]: marking

File: src/app/auth/login/index.tsx (hunk 4 above)
Comment (rachaelch3n) [inline src/app/auth/login/index.tsx]: marking

File: src/app/auth/signup/index.tsx [hunk 19]
```diff
@@ -0,0 +1,138 @@
...
+      Alert.alert('Success', 'Account created successfully!', [
+        { text: 'OK', onPress: () => router.push('/dashboard') },
+      ]);
+    } catch (error) {
+      console.error('Signup error:', error);
```
Comment (rachaelch3n) [inline src/app/auth/signup/index.tsx]: marking

**Review by rachaelch3n — CHANGES_REQUESTED** [review]
